Add final report to sytem prune command

GustavoKatel opened the pull request:
Signed-off-by: Gustavo Sampaio <gbritosampaio@gmail.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/docker/cli/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

This closes #2306
This is also my first PR, so let me know if I missed anything from the guidelines. Also let me know if I missed any test case. Thanks :D

**- What I did**

Added a final report into the `system prune` command

**- How I did it**

Refactored the prune function of each resource to return the number of deleted objects in the process.

**- How to verify it**

Run the following command: `./build/docker-linux-amd64 system prune -a --volumes -f` with some dangling resources.

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
Refactored the prune function of each resource to return the number of deleted objects in the process. Aggregated the results in a final report to be presented after the `system prune` command.

**- A picture of a cute animal (not mandatory but encouraged)**

Here is my dorky cat Penumbra.
![image](https://user-images.githubusercontent.com/1891977/73611702-2a5e8c00-45c3-11ea-8dc9-354917745cd3.png)

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/moby/moby/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "2306-system-prune-final-report" git@github.com:GustavoKatel/cli.git somewhere
$ cd somewhere
$ git rebase -i HEAD~842354555496
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

thaJeztah commented:
Sorry for the long delay; we are evaluating the UX for these commands, and looking if we can improve the output overall.

That said, we're wondering if this additional information adds much value; do you have a specific example how you use this information? (and what you use it for?)

GustavoKatel commented:
> Sorry for the long delay; we are evaluating the UX for these commands, and looking if we can improve the output overall.
> 
> That said, we're wondering if this additional information adds much value; do you have a specific example how you use this information? (and what you use it for?)

to be honest, the current output shows too much info and at the same not that much. It's mostly a huge waterfall of hashes and object ids, see:

```
# docker system prune --all --volumes -f
Deleted Networks:

---- Several lines of names omitted ----

Deleted Images:

---- Several lines of hashes omitted ----

Deleted build cache objects:

---- Several lines of object ids omitted ----

Total reclaimed space: 17.38GB
```

At the end the only thing I can managed to clearly identify is the total reclaimed space. To take a sense of anything else we need to scroll back and find a way to count/visually parse then. This is not critical ofc, but I think it's a nice addition to the output and helps you to keep an eye what's been filling up. Specially for resource constrained devices.

That said and considering this was a long time ago, maybe a better way to get that info has emerged?

Anyway, happy to work on the conflicts if you all consider this has value.
